src/runtime/GlobalObjectBuiltinRegExp.cpp

```diff
@@ -350,6 +350,135 @@ static Value builtinRegExpSplit(ExecutionState& state, Value thisValue, size_t a
     return A;
 }
 
+static Value builtinRegExpReplace(ExecutionState& state, Value thisValue, size_t argc, Value* argv, bool isNewExpression)
+{
+    ASSERT(argv != nullptr);
+    RESOLVE_THIS_BINDING_TO_OBJECT(obj, RegExp, replace);
+    RegExpObject* rx = obj->asRegExpObject();
+    String* str = argv[0].toString(state);
+    ASSERT(str != nullptr);
+    Value replaceValue = argv[1];
+    size_t lengthStr = str->length();
+    bool functionalReplace = replaceValue.isCallable();
+    bool fullUnicode = false;
+    String* replacement = String::emptyString;
+    // 7. If functionalReplace is false, then
+    if (functionalReplace == false) {
+        replaceValue = replaceValue.toString(state);
+    }
+
+    bool global = rx->option() & RegExpObject::Option::Global;
+
+    // 10. If global is true, then
+    if (global == true) {
+        fullUnicode = rx->option() & RegExpObject::Option::Unicode;
+        rx->setLastIndex(state, Value(0));
+    }
+    ValueVector results;
+    // 13. Repeat, while done is false
+    while (true) {
+        Value result = builtinRegExpExec(state, obj, 1, &argv[0], false);
+        // 13.c If result is null, set done to true
+        if (result.isNull() == true) {
+            break;
+        }
+        // 13.d
+        results.pushBack(result);
+        // 13.d.ii. If global is false, set done to true.
+        if (global == false) {
+            break;
+        }
+
+        // 13.d.iii.
+        Value matchStr = result.asObject()->get(state, ObjectPropertyName(state, Value(0))).value(state, Value(0));
+        // 13.d.iii.3 If matchStr is the empty String, then
+        if (matchStr.toString(state)->length() == 0) {
+            size_t thisIndex = rx->computedLastIndex(state);
+            size_t nextIndex = str->advanceStringIndex(thisIndex, fullUnicode);
+            rx->setLastIndex(state, Value(nextIndex));
+        }
+    }
+    //14.
+    String* accumulatedResult = String::emptyString;
+    size_t nextSourcePosition = 0;
+
+    //16. Repeat, for each result in results,
+    for (size_t i = 0; i < results.size(); i++) {
+        Value result = results[i];
+        size_t nCaptures = result.asObject()->get(state, ObjectPropertyName(state.context()->staticStrings().length)).value(state, Value(0)).toLength(state) - 1;
+
+        // Let nCaptures be max(nCaptures − 1, 0).
+        if (nCaptures == SIZE_MAX) {
+            nCaptures = 0;
+        }
+        Value matched = result.asObject()->get(state, ObjectPropertyName(state, Value(0))).value(state, Value(0));
+        size_t matchLength = matched.toString(state)->length();
+        size_t position = result.asObject()->get(state, ObjectPropertyName(state.context()->staticStrings().index)).value(state, Value(0)).asNumber();
+
+        // Let position be max(min(position, lengthS), 0).
+        if (position > lengthStr) {
+            position = lengthStr;
+        }
+
+        size_t n = 1;
+        std::vector<String*> captures;
```

Comment:
Sorry for this comment. I did't see this last time. that was my fault.
Storing String* into std::vector<GCPTR, std::allocator> causes error.
bdwgc cannot see those reference in std::vector<GCPTR, std::allocator>
You can lose String instances even if you hold on vector.
because gc may execute at line 433, 436, 452....(due to calling GC_MALLOC)
Can you store String* in GCVector?
You can reserve space If you know capture count